flow/Platform.actor.cpp

```diff
@@ -4023,13 +4029,20 @@ void setupRunLoopProfiler() {
 		// Start a thread which will use signals to log stacks on long events
 		pthread_t* mainThread = (pthread_t*)malloc(sizeof(pthread_t));
 		*mainThread = pthread_self();
-		startThread(&checkThread, (void*)mainThread, 0, "fdb-loopprofile");
+		loopProfilerThread = startThread(&checkThread, (void*)mainThread, 0, "fdb-loopprofile");
 	}
 #else
 	// No slow task profiling for other platforms!
 #endif
 }
 
+void stopRunLoopProfiler() {
```

Comment:
I think this could be called from any thread, and that means we conceivably may not even actually have initialized this state yet. Or we could be modifying it concurrently.

Reply:
What about making it mutex protected?

Reply:
I think there is still a small race here where we could first set the profiler to stopped and then skip joining the empty profiler thread, but then create the profiler thread afterward. To avoid that, I think we need to have `loopProfilerStopRequested` and `loopProfilerThread` be set/checked atomically (i.e. inside the mutexes).

Reply:
The loop profiler thread will be stopped in any case, just it may happened that it won't be joined. But this can happen when application somehow calls setupNetwork and stopNetwork concurrently from different threads. Not sure if we have a meaningful behavior for that case. If we want a meaningful behavior, then probably we should put the entire execution of setupNetwork and stopNetwork under a mutex.

Reply:
`setupRunLoopProfiler` is called from `runNetwork`, and it would be legal for someone to setup the network, run it, and quickly stop it. I doubt this will come up much, but I think the resolution here (writing `loopProfilerStopRequested` inside the mutex and checking it before starting the check thread) is simple enough to be worth avoiding this edge case

Reply:
I changed the way you proposed